compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -482,23 +482,39 @@ private BLangConstantValue calculateMod(BLangConstantValue lhs, BLangConstantVal
         return new BLangConstantValue(result, currentConstSymbol.type);
     }
 
+    private Object calculateNegationForInt(BLangConstantValue value) {
+        return -1 * ((Long) (value.value));
+    }
+
+    private Object calculateNegationForFloat(BLangConstantValue value) {
+        return String.valueOf(-1 * Double.parseDouble(String.valueOf(value.value)));
+    }
+
+    private Object calculateNegationForDecimal(BLangConstantValue value) {
+        BigDecimal valDecimal = new BigDecimal(String.valueOf(value.value), MathContext.DECIMAL128);
+        BigDecimal negDecimal = new BigDecimal(String.valueOf(-1), MathContext.DECIMAL128);
+        BigDecimal resultDecimal = valDecimal.multiply(negDecimal, MathContext.DECIMAL128);
+        return resultDecimal.toPlainString();
+    }
+
     private BLangConstantValue calculateNegation(BLangConstantValue value) {
         Object result = null;
-        switch (this.currentConstSymbol.type.tag) {
+        BType constSymbolValType = value.type;
+        int constSymbolValTypeTag = constSymbolValType.tag;
+
+        switch (constSymbolValTypeTag) {
             case TypeTags.INT:
```

Comment:
I think, we have to handle `byte` type also.

Ex:
```ballerina
const byte num1 = -(-2);
```

Reply:
created #36649 to track